core/adt/channel.hpp

```diff
@@ -108,7 +108,10 @@ namespace fc::adt {
         return false;
       }
       if (which<Handler>(state)) {
-        boost::get<Handler>(state)({});
+        try {
```

Comment:
Here was an unhandled exception in destructor.

Reply:
silently suppressing exceptions is not good.

also, there were no exception, because of type check in `if` in previous line.

may rewrite as `if (auto handler{boost::get<Handler>(&state)}) { handler({});` }

Reply:
silently ignoring exceptions is bad.
if logging `catch (...)` is difficult, maybe should add `io_context` to move potentially throwing call from destructor.
or remove `catch` and see if any handler will throw here (they should not, because they work with outcomes).

Reply:
Updated